Automattic/wp-calypso
# BackupScheduleSetting: enable fetching and updating schedule time

Related to https://github.com/Automattic/jetpack-backup-team/issues/669

## Proposed Changes

Allow users to update their daily backup time schedule. This is still under the `jetpack/backup-schedule-setting` until we decide to release it.

* Add `useScheduledTimeQuery` to handle scheduled time fetch. This one will be reused to display the scheduled time in the `Latest backup` section as part of https://github.com/Automattic/jetpack-backup-team/issues/700
* Add `useScheduledTimeMutation` to handle scheduled time update
* Enable `jetpack/backup-schedule-setting` on Jetpack Cloud development and horizon (live branches) environments

## Demo

https://github.com/user-attachments/assets/c5b08c2b-3697-40b7-8c36-c9f0a7d2ebff

## Why are these changes being made?


https://github.com/Automattic/jetpack-backup-team/issues/669

## Testing Instructions



* Spin up a Jetpack Cloud live branch
* Pick a site with Jetpack VaultPress Backup plan
* Navigate to the Settings page
* You should see the `Daily backup time schedule` setting at the bottom of the page
* If you never updated your schedule time, you should see a default one with the `Default time` message
* Pick another time slot on the dropdown.
  * The dropdown should be in your local time (the one configured on your site). So if you click 2:00pm (your local time), it will be stored in UTC.
  * When you click it, it will trigger the update (you should see the dropdown being disabled for a bit)
  * Once it completes, you should see a  success notice with the message `Daily backup time successfully changed.`
  * The dropdown should be updated with the new value you selected.
  * The message below the dropdown should be changed to `Time set by <user>. UTC: X:XX - X:XX XX`.
    * <user> will be the name of the user who performed the action
    * `UTC: X:XX - X:XX XX` will be the UTC time of the schedule time saved.
## Pre-merge Checklist
